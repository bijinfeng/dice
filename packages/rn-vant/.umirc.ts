import { defineConfig, IConfig } from 'dumi';

const isDev = process.env.NODE_ENV === 'development';
const isProd = process.env.NODE_ENV === 'production' && process.env.PREVIEW_PR !== 'true';

export default defineConfig({
  exportStatic: isProd ? {} : false,
  title: 'React Native Vant',
  mode: 'site',
  algolia: {
    appId: 'TM4VZTMTDL',
    apiKey: 'c0a3916beb89f5500d092ab4e19143b6',
    indexName: 'docsearch',
  },
  favicon: 'https://img01.yzcdn.cn/vant/logo.png',
  logo: 'https://img01.yzcdn.cn/vant/logo.png',
  themeConfig: {
    demoUrl: isDev ? 'http://localhost:19006' : 'https://bijinfeng.github.io/rn-vant/example',
  },
  navs: [
    null,
    {
      logo: 'https://b.yzcdn.cn/vant/logo/github.svg',
      path: 'https://github.com/bijinfeng/rn-vant',
    },
  ],
  resolve: {
    passivePreview: true,
  },
  hash: isProd,
  base: isProd ? '/rn-vant/' : '/',
  publicPath: isProd ? 'https://cdn.jsdelivr.net/gh/bijinfeng/rn-vant@gh-pages/' : '/',
  sitemap: {
    hostname: 'https://bijinfeng.github.io/rn-vant/',
  }
  // more config: https://d.umijs.org/config
} as IConfig);
