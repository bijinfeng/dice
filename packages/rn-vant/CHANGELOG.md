# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.2](https://github.com/bijinfeng/rn-vant/compare/rn-vant@0.1.0...rn-vant@0.0.2) (2022-05-14)

**Note:** Version bump only for package rn-vant





# 0.1.0 (2022-05-14)


### Features

* change package name ([2851a86](https://github.com/bijinfeng/rn-vant/commit/2851a864a0842a035b3f9affd1deb92496071a4c))
