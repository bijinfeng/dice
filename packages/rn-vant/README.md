# `rn-vant`

> TODO: description

## Usage

```
const diceUi = require('rn-vant');

// TODO: DEMONSTRATE API
```
