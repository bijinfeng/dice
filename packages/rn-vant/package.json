{
  "name": "rn-vant",
  "version": "0.1.0",
  "description": "React Native UI Components",
  "keywords": [
    "react-native",
    "Components"
  ],
  "author": "bijinfeng <bijinfeng@bytedance.com>",
  "license": "MIT",
  "main": "lib/commonjs/index",
  "module": "lib/module/index",
  "types": "lib/typescript/index.d.ts",
  "react-native": "src/index",
  "source": "src/index",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib",
    "src",
    "docs"
  ],
  "scripts": {
    "build": "bob build",
    "site": "dumi dev",
    "build:site": "dumi build"
  },
  "dependencies": {
    "@callstack/react-theme-provider": "^3.0.6",
    "@gorhom/portal": "^1.0.12",
    "@ptomasroos/react-native-multi-slider": "^2.2.2",
    "@rn-vant/icons": "^0.1.0",
    "color": "^3.1.2",
    "lodash-es": "^4.17.21",
    "react-native-gesture-handler": "^2.4.2",
    "react-native-iphone-x-helper": "^1.3.1",
    "react-native-svg": "^12.1.1",
    "react-native-swiper": "^1.6.0"
  },
  "devDependencies": {
    "@rn-vant/dumi-theme-vant": "^0.1.0",
    "@types/color": "^3.0.2",
    "@types/lodash-es": "^4.17.6",
    "@types/react-native": "^0.65.5",
    "dumi": "^1.1.37",
    "react-native-builder-bob": "^0.18.2",
    "typescript": "^4.4.3"
  },
  "peerDependencies": {
    "react": "*",
    "react-native": "*"
  },
  "react-native-builder-bob": {
    "source": "src",
    "output": "lib",
    "targets": [
      "commonjs",
      "module",
      [
        "typescript",
        {
          "project": "tsconfig.build.json"
        }
      ]
    ]
  },
  "gitHead": "38c04917b2fa4ecc4227d7f3e201786b4e0e48f0"
}
