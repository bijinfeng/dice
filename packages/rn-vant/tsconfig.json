{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "paths": {
      "rn-vant": ["./src/index"],
    },
  },
  "exclude": ["lib/**/*"]
}