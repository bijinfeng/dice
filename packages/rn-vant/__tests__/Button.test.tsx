import * as React from 'react';
import renderer from 'react-test-renderer';
import { Button } from 'rn-vant';

it('test snapshot', () => {
  const component = renderer.create(<Button>default button</Button>);

  expect(component.toJSON()).toMatchSnapshot();
});
