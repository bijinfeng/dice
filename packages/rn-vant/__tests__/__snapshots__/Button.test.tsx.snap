// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#dcdee0",
      "borderRadius": 2,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "paddingHorizontal": 15,
      "position": "relative",
    }
  }
>
  <Text
    numberOfLines={1}
    selectable={false}
    style={
      Object {
        "color": "#323233",
        "fontSize": 14,
      }
    }
  >
    default button
  </Text>
</View>
`;
